src/specular/sequencing/mod.rs

```diff
@@ -11,15 +16,23 @@ use crate::{
     l1::L1BlockInfo,
 };
 
+use crate::specular::common::{
+    SetL1OracleValuesInput, SET_L1_ORACLE_VALUES_ABI, SET_L1_ORACLE_VALUES_SELECTOR,
+};
+
 pub mod config;
 
 pub struct AttributesBuilder {
     config: config::Config,
+    client: Option<SignerMiddleware<Provider<Http>, LocalWallet>>,
```

Comment:
why `Option`?

Reply:
its for `test_is_ready` to run

Reply:
hm i prefer using a `impl trait`/generic arg or `Box<dyn trait>`

Reply:
fixed with generics, looks much uglier lol